tests/cnf/core/network/sriov/tests/metricsExporter.go

```diff
@@ -0,0 +1,310 @@
+				WithRDMA(true)
+		}
+	}
+
+	return policy
+}
+
+func defineNetwork(role, devType string) *sriov.NetworkBuilder {
+	network := sriov.NewNetworkBuilder(
+		APIClient, role+devType, NetConfig.SriovOperatorNamespace, tsparams.TestNamespaceName, role+devType).
+		WithMacAddressSupport().
+		WithIPAddressSupport().
+		WithStaticIpam()
+
+	return network
+}
+
+func definePod(role, devType, worker string) *pod.Builder {
+	var podbuild *pod.Builder
+
+	var netAnnotation []*types.NetworkSelectionElement
+
+	switch role {
+	case "client":
+		netAnnotation = []*types.NetworkSelectionElement{
+			{
+				Name:       role + devType,
+				MacRequest: tsparams.ClientMacAddress,
+				IPRequest:  []string{tsparams.ClientIPv4IPAddress},
+			},
+		}
+	case "server":
+		netAnnotation = []*types.NetworkSelectionElement{
+			{
+				Name:       role + devType,
+				MacRequest: tsparams.ServerMacAddress,
+				IPRequest:  []string{tsparams.ServerIPv4IPAddress},
+			},
+		}
+	}
+
+	podbuild = pod.NewBuilder(APIClient, role+"pod", tsparams.TestNamespaceName, NetConfig.CnfNetTestContainer).
+		WithNodeSelector(map[string]string{"kubernetes.io/hostname": worker}).
+		WithSecondaryNetwork(netAnnotation)
+
+	return podbuild
+}
+
+func createTestResources(client, server testResource) (*pod.Builder, *pod.Builder) {
+	var err error
+
+	By("Creating SriovNetworkNodePolicy for client Pod")
+
+	err = sriovenv.CreateSriovPolicyAndWaitUntilItsApplied(client.policy, tsparams.MCOWaitTimeout)
+	Expect(err).ToNot(HaveOccurred(), "Failed to configure client SR-IOV policy")
+
+	By("Creating SriovNetworkNodePolicy for server Pod")
+
+	err = sriovenv.CreateSriovPolicyAndWaitUntilItsApplied(server.policy, tsparams.MCOWaitTimeout)
+	Expect(err).ToNot(HaveOccurred(), "Failed to configure server SR-IOV policy")
+
+	By("Creating SriovNetwork for client Pod")
+
+	_, err = client.network.Create()
+	Expect(err).ToNot(HaveOccurred(), "Failed to create client SR-IOV network")
+
+	By("Creating SriovNetwork for server Pod")
+
+	_, err = server.network.Create()
+	Expect(err).ToNot(HaveOccurred(), "Failed to create server SR-IOV network")
+
+	By("Creating client Pod")
+
+	createdClientPod, err := client.pod.Create()
+	Expect(err).ToNot(HaveOccurred(), "Failed to create client pod")
+
+	By("Creating server Pod")
+
+	createdServerPod, err := server.pod.Create()
+	Expect(err).ToNot(HaveOccurred(), "Failed to create server pod")
+
+	return createdClientPod, createdServerPod
+}
+
+func execPromQLandReturnString(query []string) string {
+	By("Running PromQL to fetch metrics of serverpod")
+
+	promPods, err := pod.List(APIClient, "openshift-monitoring", metav1.ListOptions{LabelSelector: "prometheus=k8s"})
+	Expect(err).ToNot(HaveOccurred(), "Failed to get prometheus pods")
+
+	glog.V(90).Infof(fmt.Sprintf("Running PromQL query: %s", query))
+	output, err := promPods[0].ExecCommand(query, "prometheus")
+	Expect(err).ToNot(HaveOccurred(), "Failed to get promQL output from prometheus pod")
```

Comment:
Please share output.String()  as well

Reply:
I am loggin output.String() already. if thats what you meant ?
glog.V(90).Infof(fmt.Sprintf("Received PromQL output: %s", output.String()))

Reply:
But it's after the error check. If we get  an error, we'll not reach log

Reply:
ok. included this now.

`Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("Failed to get promQL output from prometheus pod. Output: %s", output.String()))`